src/Storages/MergeTree/MergeTreeIndexText.h

```diff
@@ -128,44 +139,39 @@ struct PostingsSerialization
         RawPostings = 1ULL << 0,
         /// If set, the posting list is embedded into the dictionary block to avoid additional random reads from disk.
         EmbeddedPostings = 1ULL << 1,
+        /// If set, the posting list is serialized as a single block.
+        SingleBlock = 1ULL << 2,
     };
 
-    static UInt64 serialize(UInt64 header, PostingListBuilder && postings, WriteBuffer & ostr);
-    static PostingListPtr deserialize(UInt64 header, UInt32 cardinality, ReadBuffer & istr);
+    static void serialize(PostingListBuilder & postings, UInt64 header, WriteBuffer & ostr);
+    static void serialize(PostingList & postings, UInt64 header, WriteBuffer & ostr);
+    static void serialize(const roaring::api::roaring_bitmap_t & bitmap, WriteBuffer & ostr);
```

Comment:
Could we use the `PostingList` typedef instead of `roaring::api::roaring_bitmap_t`?

Reply:
Yes, that makes sense.